src/kudu/consensus/raft_consensus.cc

```diff
@@ -3052,6 +3061,9 @@ Status RaftConsensus::SetCurrentTermUnlocked(int64_t new_term,
   if (flush == FLUSH_TO_DISK) {
     CHECK_OK(cmeta_->Flush());
   }
+  // Trigger term advancement callback
+  ScheduleTermAdvancementCallback(new_term);
```

Comment:
TermAdvancementCallback should probably be scheduled after calling ClearLeaderUnlocked().

Reply:
I don't think it would make any difference after we have flushed. Flushing to disk is the turning point. Let me know if you think the order would affect correctness.

Reply:
Just that there might be a race between clearing the leader uuid and the callback getting scheduled (in case the callback has a business logic of fetching the leader_uuid). In general, it feels like a good practice to guarantee that the registered callbacks will be called either before making any changes OR after making all the local changes (and in this case we would callback is fired after making all the changes locally)